synapseclient/core/async_utils.py

```diff
@@ -93,6 +93,65 @@ def wrap_async_to_sync(coroutine: Coroutine[Any, Any, Any], syn: "Synapse") -> A
         return asyncio.run(coroutine)
 
 
+def wrap_async_generator_to_sync_generator(async_gen_func: Callable, *args, **kwargs):
+    """
+    Wrap an async generator function to be called in a sync context, returning a sync generator.
+    This function takes an async generator function and its arguments, then yields items
+    synchronously by running the async generator in the appropriate event loop.
+    Arguments:
+        async_gen_func: The async generator function to wrap
+        *args: Positional arguments to pass to the async generator function
+        **kwargs: Keyword arguments to pass to the async generator function
+    Yields:
+        Items from the async generator, yielded synchronously
+    """
+    loop = None
+
+    try:
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        pass
+
+    if loop:
+        nest_asyncio.apply(loop=loop)
+
+        # Create the async generator
+        async_gen = async_gen_func(*args, **kwargs)
+
+        # Yield items from the async generator synchronously
+        try:
+            while True:
+                try:
+                    item = loop.run_until_complete(async_gen.__anext__())
+                    yield item
+                except StopAsyncIteration:
```

Comment:
Could this try/except loop break for other reasons except `StopAsyncIteration`? Does it matter that there's no cleanup after the break here?